hphp/util/hash-crc-arm.S

```diff
@@ -0,0 +1,189 @@
+#include "hphp/util/etch-helpers.h"
+#include "hphp/util/hphp-config.h"
+
+#if defined(ENABLE_AARCH64_CRC) && !defined(NO_HWCRC) && \
+    !defined(__CYGWIN__) && !defined(__MINGW__) && !defined(_MSC_VER)
+
+.file   "hphp/util/hash-crc-arm.S"
+
+#define HASH_FUNC_NAME _ZN4HPHP20hash_string_i_unsafeEPKcj
+
+ETCH_SECTION(HASH_FUNC_NAME)
+.globl    HASH_FUNC_NAME
+ETCH_TYPE(HASH_FUNC_NAME, @function)
+ETCH_ALIGN16
+ETCH_NAME(HASH_FUNC_NAME):
+        CFI(startproc)
+        negs    w1, w1
+        mov     w9, #0xffffffff
+        beq     ETCH_LABEL(iend)
+        mov     w3, w1
+        mov     x4, #0xdfdfdfdfdfdfdfdf
+        b       ETCH_LABEL(iheader)
+
+ETCH_ALIGN16
```

Comment:
Looking into GCC's tuning flags, I notice the following:
Some loop tunings are 16 byte aligned and others are 8 byte aligned.  I know there was some discussion on the GCC mailing list about this too.  For ThunderX even though it is 8 byte aligned for gcc, the fetcher happens on a 16 byte boundary.  What I found for ThunderX was for some benchmarks 16 byte was better than 8 bytes but I did not dig far enough into why.  I think 16 byte is reasonable here and might only cause a few cycle hit on some cores.